airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/DogStatsDMetricClient.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
+import com.timgroup.statsd.StatsDClient;
+import io.airbyte.config.Configs;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Light wrapper around the DogsStatsD client to make using the client slightly more ergonomic.
+ * <p>
+ * This class mainly exists to help Airbyte instrument/debug application on Airbyte Cloud. The
+ * methods here do not fail loudly to prevent application disruption.
+ * <p>
+ * Open source users are free to turn this on and consume the same metrics.
+ * <p>
+ * This class is intended to be used in conjection with {@link Configs#getPublishMetrics()}.
+ */
+@Slf4j
+public class DogStatsDMetricClient implements MetricClient {
+
+  private boolean instancePublish = false;
+  private StatsDClient statsDClient;
+
+  /**
+   * Traditional singleton initialize call. Please invoke this before using any methods in this class.
+   * Usually called in the main class of the application attempting to publish metrics.
+   */
+  public void initialize(final MetricEmittingApp app, final DatadogClientConfiguration config) {
+    if (statsDClient != null) {
+      throw new RuntimeException("You cannot initialize configuration more than once.");
+    }
+
+    if (!config.publish) {
+      // do nothing if we do not want to publish. All metrics methods also do nothing.
+      return;
+    }
+
+    log.info("Starting DogStatsD client..");
+    instancePublish = config.publish;
+    statsDClient = new NonBlockingStatsDClientBuilder()
+        .prefix(app.getApplicationName())
+        .hostname(config.ddAgentHost)
+        .port(Integer.parseInt(config.ddPort))
+        .build();
+  }
+
+  @VisibleForTesting
+  public synchronized void flush() {
```

Comment:
```suggestion
  public synchronized void shutdown() {
```

`shutdown` seems like a better name.

Reply:
add a javadoc to make it clear this clears the underlying publishing clients.

Reply:
thinking out loud - should `initialize` and `shutdown` be part of the interface? the factory can then call the underlying client's initialise and shutdown methods if needed.

Reply:
make sense... I was implementing OTEL metrics and I need to have shutdown method called from here, otherwise the OTEL will not be reset correctly.

Reply:
I'll leave initialize out because I'm not sure what OTEL client will need for initialize. Will add it back if needed in the next PR.